### PR TITLE
Add Processing entity

### PR DESCRIPTION
**Pull request by PapsOu**
### Summary

Split PIA in two separate entities in order to provide a dedicated Processing management.

### CODE REVIEW

File: src/Entity/Pia/Folder.php [hunk 1]
```diff
@@ -107,6 +107,15 @@ class Folder implements Timestampable
      */
     protected $pias;
```
Comment (RomainSanchez) [inline src/Entity/Pia/Folder.php]: Folder must no longer reference pias

File: src/Entity/Pia/ProcessedDataType.php [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+{
+    use ResourceTrait;
+
+    /**
+     * @ORM\Column(type="string")
+     * @JMS\Groups({"Default", "Export"})
+     *
+     * @var string
+     */
+    protected $specificDataRetentionPeriod;
```
Comment (RomainSanchez) [inline src/Entity/Pia/ProcessedDataType.php]: retentionPeriod?

File: src/Entity/Pia/ProcessedDataType.php [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+     */
+    public function setSpecificDataRetentionPeriod(string $specificDataRetentionPeriod): void
+    {
+        $this->specificDataRetentionPeriod = $specificDataRetentionPeriod;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isSensitiveData(): bool
```
Comment (RomainSanchez) [inline src/Entity/Pia/ProcessedDataType.php]: isSensitive?

File: src/Entity/Pia/ProcessedDataType.php [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    {
+        return $this->sensitiveData;
+    }
+
+    /**
+     * @param bool $sensitiveData
+     */
+    public function setSensitiveData(?bool $sensitiveData = true): void
```
Comment (RomainSanchez) [inline src/Entity/Pia/ProcessedDataType.php]: setIsSensitive

File: src/Entity/Pia/ProcessedDataType.php (hunk 2 above)
Comment (PapsOu) [reply src/Entity/Pia/ProcessedDataType.php]: https://translate.google.com/?oe=utf-8&client=firefox-b-ab&um=1&ie=UTF-8&hl=fr&client=tw-ob#en/fr/data%20retention%20period :wink:
The translation may be wrong...

File: src/Entity/Pia/ProcessedDataType.php (hunk 4 above)
Comment (PapsOu) [reply src/Entity/Pia/ProcessedDataType.php]: setSensitive instead ?

File: src/Migrations/Version20180731132909.php [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        $this->addSql('ALTER TABLE pia_processing DROP data_medium_description');
+        $this->addSql('ALTER TABLE pia_processing DROP standards_description');
+        $this->addSql('ALTER TABLE pia_processing RENAME COLUMN data_transfer_outside_eu TO non_eu_transfer');
+    }
+
+    public function down(Schema $schema)
+    {
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'postgresql', 'Migration can only be executed safely on \'postgresql\'.');
+
+        $this->addSql('CREATE SCHEMA public');
```
Comment (GlennCavarle) [inline src/Migrations/Version20180731132909.php]: do we need that ?

**Pull request merged**